### PR TITLE
Validate docs (plus)

### PR DESCRIPTION
**Pull request by joshmoore**
* [x] schema: add README with file list
 * [x] validation: add examples to module; add usage page
 * [x] misc: update table-of-contents ordering
 * [x] misc: include license file
 * [x] misc: fix codebook and intensity_tables autodocs
 * [x] misc: move to `index.rst` in several cases (**see discussion**)
 * [x] sphinx: add `program-ouputs` for keeping `--help` uptodate

Note: this doesn't yet include enforcing validation via `STARFISH_VALIDATION_STRICT`. That's waiting on gh-789

### CODE REVIEW

**Comment by joshmoore** [comment]
For discussion: I'd propose that we move to `topic/index.rst` as frequently as possible since it produces nice URLs like https://spacetx-starfish.readthedocs.io/en/latest/API/image/ rather than https://spacetx-starfish.readthedocs.io/en/latest/API/image/filtering/filtering.html

File: docs/source/index.rst [hunk 1]
```diff
@@ -19,22 +19,22 @@ Starfish documentation
...
-    license/license.rst
+    contributing/contributing.md
 
 .. toctree::
-    sptx-format/index.rst
+    debugging.rst
 
 .. toctree::
-    usage/index.rst
+    glossary/glossary.md
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/index.rst]: I was actually surprised that sphinx handled the mix of rst and md so well. I don't feel a particular need to unify on rst (I find markdown easier to author myself) but where we want advanced sphinx features, we'll likely have to convert.

File: docs/source/index.rst [hunk 2]
```diff
@@ -19,22 +19,22 @@ Starfish documentation
...
-    contributing/contributing.md
+    usage/index.rst
 
 .. toctree::
-    debugging.rst
+    sptx-format/index.rst
 
 .. toctree::
-    license/license.rst
+    contributing/contributing.md
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/index.rst]: > Are these links screwy?

You mean why has this change been made? Just a change of ordering.

> Do you have any suggestions about how to deal with cases where

The literal include and markdown includes seem to work with relative links. in other words, I don't see an issue. I can push a fish for gh-883.

File: docs/source/API/validation/index.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            Subschema level:        0
+            Path to error:          required
+            Filename:               ../field_of_view/field_of_view.json
+    
+      warnings.warn(message)
+
+
+This message tells you which schema has failed validation (fov-manifest.json), what type of error
+has been encountered (a required field is missing), and the name of the file which is invalid
+(field_of_view.json) if it has been provided. Validation of json objects will not provide a filename.
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/API/validation/index.rst]: screen shot 2018-12-18 at 16 20 09

File: docs/source/license/index.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.. _license:
+
+License
+=======
+
+.. literalinclude:: ../../../LICENSE
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/license/index.rst]: But at the moment, you don't see anything that is rendering incorrectly in this PR, correct?

[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply docs/source/license/index.rst]: Nope, just pointing out that it's a trap that we'll eventually want to fix.

File: docs/source/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply docs/source/index.rst]: Ok, great.

File: docs/source/API/validation/index.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply docs/source/API/validation/index.rst]: 👍

[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/API/validation/index.rst]: Pushed.

File: docs/source/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/index.rst]: Pushed.

File: docs/source/conf.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 dir_, _ = os.path.split(__file__)
 root_dir = os.path.abspath(os.path.join(dir_, '..', '..'))
 sys.path.insert(0, root_dir)
+print(sys.path)
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/conf.py]: Nope. Already removed though in the merged gh-789

File: docs/source/usage/fov-builder/fov-builder.rst [hunk 6]
```diff
@@ -0,0 +1,35 @@
+.. _cli_build:
+
+Synthetic experiments
+=====================
+
+Building Synthetic SpaceTx-specification compliant experiments
```
[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/source/usage/fov-builder/fov-builder.rst]: will push to gh-904

**Pull request merged**